### PR TITLE
Add tests suite

### LINKED ISSUE #8: Addition of tests to check build status
The package currently has no test coverage at all. There is thus no method of checking build status with CI/CD.
The core functionality at least should have test coverage.

### PR DESCRIPTION
**Pull request by lahdjirayhan**
This PR is currently working in progress, to fix #8.

List of public methods to write tests of:

- [x] `get_city_air`
- [x] `get_city_forecast`
- [x] `get_city_station_options`
- [x] `get_coordinate_air`
- [x] `get_historical_data`
- [x] `get_multiple_city_air`
- [x] `get_multiple_coordinate_air`
- [x] `get_range_coordinates_air`
- [x] `get_specific_parameter`
- [ ] `reset_token` ?

In summary:
- use `pytest` as the test framework
- use `pytest-recording` plugin to utilize `vcr.py` for mocking API response data
- one test file per public method

This PR should also:
- [x] add instructions to run tests (preferably in CONTRIBUTING)
- [x] add guidelines to add or modify tests, esp. about when will it be necessary to do so
- [x] add guidelines about VCR.py cassettes

Potential discussion points:
- Should we use `tox` for doing the tests? Especially in light of error discovered in #119

### CODE REVIEW

**Comment by Milind220** [comment]
Great PR! Gonna be really awesome to have a testing suite, and this one looks really comprehensive :)

> Should we use tox for doing the tests? Especially in light of error discovered in https://github.com/Milind220/Ozone/issues/119

I'm going to read up on tox, seeing as I don't currently know what it is. I'm assuming it helps catch errors that may crop up in the process of packaging and importing?

**Comment by lahdjirayhan** [comment]
Some caveats about the tests and cassettes:
1. The tests are not written in a way that current Ozone will just pass. I've used my best judgments when exploring/writing the tests. It feels different when writing tests vs when writing code.
2. Some cassettes are (unluckily? luckily?) getting some quirky edge cases that throws error. I've never encountered this behavior in particular: Paris' AQI is given out as `"-"` a string, with just a hyphen in it. I get that it's supposed to mean NA, maybe, but it throws a `ValueError: could not convert string to float: '-'` that I've never seen before.
3. There is an uncaught bug in current Ozone! Try using `get_range_coordinates_air`, and you'll see (I compare it with `v1.5.0` tag, there was no error there). This error is likely missed when refactoring the code.

My suggestion to go about this:

1. I recommend fixing Ozone to pass the tests than the other way around. They're mostly minor aspects, but please do check the failing tests anyway.
2. I recommend fixing Ozone to handle such edge cases, and leave the cassettes as it is (not re-record). Let's just say we're lucky to have an edge case in our cassette. But re-recording them would be fine anyway.
3. This has to be fixed asap.

---

@Milind220 Given we're both new to tox, I think we should focus on the test suite as it is and think about it in another time. Sorry for raising it earlier, it felt like I've bothered you and overcomplicated the PR unnecessarily. 🙏🏼

**Comment by Milind220** [comment]
> 3. There is an uncaught bug in current Ozone! Try using `get_range_coordinates_air`, and you'll see (I compare it with `v1.5.0` tag, there was no error there). This error is likely missed when refactoring the code.

Oh yikes! Great that we have tests now though :)

> My suggestion to go about this:
> 
> 1. I recommend fixing Ozone to pass the tests than the other way around. They're mostly minor aspects, but please do check the failing tests anyway.

I absolutely agree, tests will ensure that Ozone's code is up to a decent standard.

> 2. I recommend fixing Ozone to handle such edge cases, and leave the cassettes as it is (not re-record). Let's just say we're lucky to have an edge case in our cassette. But re-recording them would be fine anyway.
> 3. This has to be fixed asap.

Let's leave the cassettes as they are, and fix the bugs asap for sure. 

> @Milind220 Given we're both new to tox, I think we should focus on the test suite as it is and think about it in another time. Sorry for raising it earlier, it felt like I've bothered you and overcomplicated the PR unnecessarily. 🙏🏼

No worries, we can always add it in another time.

Awesome work! I went through the tests and they look really solid and well formulated. Can't wait to see Ozone'd code quality improve with the tests checking :)

**Review by Milind220 — APPROVED** [review]
Tests look really great - I think we'll find a lot more wrong with Ozone - which is a good thing!

File: tests/test_get_range_coordinates_air.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        LOWER_BOUND, UPPER_BOUND, params=[BAD_PARAM_NAME]
+    )
+    param_value = result.at[0, BAD_PARAM_NAME]
+    assert numpy.isnan(param_value)
+
+
+@pytest.mark.vcr
+def test_bad_coordinates():
+    with pytest.raises(Exception):
+        api.get_range_coordinates_air(lower_bound=("lol", "bruh"), upper_bound=(0, 0))
```
Comment (Milind220) [inline tests/test_get_range_coordinates_air.py:91]: hahahaha 'lol' and 'bruh'

**Comment by Milind220** [comment]
@lahdjirayhan Just to confirm, this PR is complete right? I do have a couple questions.

* What is the current mechanism to run these tests? Do we run them on locally on our own computers? 
* We still have to add continuous testing integration right?

File: tests/test_get_range_coordinates_air.py (hunk 1 above)
Comment (lahdjirayhan) [reply tests/test_get_range_coordinates_air.py:91]: - Short enough
- Not interpretable as a valid coordinate value

 So ... hehe.
